sycl/include/CL/sycl/platform.hpp

```diff
@@ -48,22 +58,50 @@ class platform {
 
   bool operator!=(const platform &rhs) const { return !(*this == rhs); }
 
-  cl_platform_id get() const { return impl->get(); }
-
-  bool has_extension(const string_class &extension_name) const {
-    return impl->has_extension(extension_name);
-  }
-
-  bool is_host() const { return impl->is_host(); }
-
+  /// Returns an OpenCL interoperability platform.
+  ///
+  /// @return an instance of OpenCL cl_platform_id.
+  cl_platform_id get() const;
+
+  /// Checks if platform supports specified extension.
+  ///
+  /// @param ExtensionName is a string containing extension name.
+  /// @return true if specified extension is supported by this SYCL platform.
+  bool has_extension(const string_class &ExtensionName) const;
+
+  /// Checks if this SYCL platform is a host platform.
+  ///
+  /// @return true if this SYCL platform is a host platform.
+  bool is_host() const;
+
+  /// Returns all SYCL devices associated with this platform.
+  ///
+  /// If this SYCL platform is a host platform, resulting vector contains only
+  /// a single SYCL host device. If there are no devices that match given device
+  /// type, resulting vector is empty.
```

Comment:
What will be returned if GPU device type is requested from a SYCL host platform?
"a single SYCL host device" or "resulting vector is empty"?